src/QueryRenderer.ts

```diff
@@ -206,6 +212,40 @@ class QueryRenderChild extends MarkdownRenderChild {
         });
     }
 
+    private static addGroupHeadings(
+        content: HTMLDivElement,
+        groupHeadings: GroupHeading[],
+    ) {
+        for (const heading of groupHeadings) {
+            QueryRenderChild.addGroupHeading(content, heading);
+        }
+    }
+
+    private static addGroupHeading(
+        content: HTMLDivElement,
+        group: GroupHeading,
+    ) {
+        let header: any;
+        // Is it possible to remove the repetition here?
+        // Ideally, by creating a variable that contains h4, h5 or h6
+        // and then only having one call to content.createEl().
+        if (group.nestingLevel == 0) {
+            header = content.createEl('h4', {
+                cls: 'tasks-group-heading',
+            });
+        } else if (group.nestingLevel == 1) {
+            header = content.createEl('h5', {
+                cls: 'tasks-group-heading',
+            });
+        } else {
+            // Headings nested to 2 or more levels are all displayed with 'h6:
+            header = content.createEl('h6', {
+                cls: 'tasks-group-heading',
+            });
+        }
+        header.appendText(group.name);
```

Comment:
Thanks. That's pretty much what I had originally, and it gives:
`TS2345: Argument of type 'string' is not assignable to parameter of type 'keyof HTMLElementTagNameMap'`
I've just noted the error message in the function, in case anyone else can see a way to improve it.

Reply:
In that case, the type of `headerLevel` should not be `string`, but `keyof HTMLElementTagNameMap`. That also prevents you from assigning non-sense in the switch/case (like `'foo'`).

Reply:
> In that case, the type of headerLevel should not be string, but keyof HTMLElementTagNameMap. That also prevents you from assigning non-sense in the switch/case (like 'foo').

Noted. Thank you. Will look at this later, after the merge.